code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -156,8 +156,60 @@
 		//G.reagents.trans_to(src,G.amount_per_transfer_from_this)
 	else if(istype(O,/obj/item/weapon/grab))
 		var/obj/item/weapon/grab/G = O
-		to_chat(user, "<span class='alert'>This is ridiculous. You can not fit \the [G.affecting] in this [src].</span>")
-		return 1
+		if(!istype(src, /obj/machinery/kitchen_machine/oven) && !istype(src, /obj/machinery/kitchen_machine/grill))
```

Comment:
@Twinmold93 istype(src) is disgustingly breaking all semblance of object oriented programming, that's why I left eyes on all of them; These should all be on their respective machine's /attackby()